adding google cloud storage upload (#747)
codemagic-ci-cd/codemagic-docs: content/yaml-publishing/google-cloud-storage.md

---
title: Google Cloud Storage
description: How to publish build artifacts to Google Cloud Storage using codemagic.yaml
weight: 14
---

In order to publish your generated artifacts to Google Cloud Storage:

> 1. Create a new Service Account and generate a JSON file with **Storage Object Creator** role. Save your JSON file to [environment variable](https://docs.codemagic.io/variables/environment-variable-groups/). 

Reviewer: Can we give a bit more detail around this?

Maybe in the first step explain how to create the required service account as follows:

Log in to your Google Cloud platform [console](https://console.cloud.google.com/).

Create a new service account under IAM & Admin > Service Accounts
Give Storage Object Admin access to the newly created service account.
Create a key for the service account. Choose key type as JSON. This will cause a JSON key file to be downloaded.

Then in another step, say that they should create an environment variable called GCLOUD_STORAGE_KEY and save the contents of the Service Account JSON to the the variable. Then link to the docs as you have done already.

> 2. Go to your Google Cloud Platfrom console and open Cloud Storage, create a new bucket and use the exact same bucket name in the following request.

Reviewer: ```suggestion
2. Go to your Google Cloud Platform console and open Cloud Storage. Make a note of the name of the bucket you want to upload to or create a new bucket. 
```

> 3. Use the following request in your post-publishing script:

Reviewer: ```suggestion
3. Use the following script in your post-publishing script or create a new script step if you are using codemagic.yaml:
```

> ```
> echo $GCLOUD_STORAGE_KEY > $FCI_BUILD_DIR/gcloud.json
> gcloud auth activate-service-account --key-file $FCI_BUILD_DIR/gcloud.json
> gsutil cp $FCI_BUILD_DIR/app/build/outputs/**/*.apk gs://YOUR_BUCKET_NAME
> ```

Reviewer: ```suggestion
echo $GCLOUD_STORAGE_KEY > $FCI_BUILD_DIR/gcloud_storage_key.json
gcloud auth activate-service-account --key-file $FCI_BUILD_DIR/gcloud_storage_key.json
gsutil cp $FCI_BUILD_DIR/app/build/outputs/**/*.apk gs://YOUR_BUCKET_NAME
```

> 1. Log in to your Google Cloud platform [console](https://console.cloud.google.com/). Create a new service account under **IAM & Admin > Service Accounts**.Find more information about how to create a service account [here](https://docs.codemagic.io/knowledge-base/google-services-authentication/#creating-a-service-account). Give **Storage Object Admin** access to the newly created service account which needs to be followed by generating a key as JSON for the service account. This will cause a JSON key file to be downloaded. As the last step in the process, you should create an environment variable called **GCLOUD_STORAGE_KEY** and save the contents of the Service Account JSON to the the variable. More information about environment variable groups can be found [here](https://docs.codemagic.io/variables/environment-variable-groups/). 

Reviewer: ```suggestion
1. Log in to your Google Cloud platform [console](https://console.cloud.google.com/).
 
2. Create a new service account under **IAM & Admin > Service Accounts** and give **Storage Object Admin** access to the newly created service account. Find more information about how to create a service account [here](https://docs.codemagic.io/knowledge-base/google-services-authentication/#creating-a-service-account).

3. Generate a key as JSON for the service account. This will cause a JSON key file to be downloaded. 

4. In the Codemagic web app you should create an environment variable called **GCLOUD_STORAGE_KEY** and add the contents of the Service Account JSON as its value. More information about environment variable groups can be found [here](https://docs.codemagic.io/variables/environment-variable-groups/). 
```

> 2. Go to your Google Cloud Platform console and open Cloud Storage. Make a note of the name of the bucket you want to upload to or create a new bucket.

Reviewer: ```suggestion
5. Go to your Google Cloud Platform console and open Cloud Storage. Make a note of the name of the bucket you want to upload to or create a new bucket.
```

> 3. Use the following script in your post-publishing script or create a new script step if you are using codemagic.yaml:

Reviewer: ```suggestion
6. Use the following script in your post-publishing script or create a new script step if you are using codemagic.yaml:
```

---
title: Google Cloud Storage
description: How to publish build artifacts to Google Cloud Storage using codemagic.yaml
weight: 14
---

In order to publish your generated artifacts to Google Cloud Storage:

1. Log in to your Google Cloud platform [console](https://console.cloud.google.com/). 

2. Create a new service account under **IAM & Admin > Service Accounts**.Find more information about how to create a service account [here](https://docs.codemagic.io/knowledge-base/google-services-authentication/#creating-a-service-account). 

3. Generate a key as JSON for the service account. This will cause a JSON key file to be downloaded. 

4. In the Codemagic web app you should create an environment variable called **GCLOUD_STORAGE_KEY** and add the contents of the Service Account JSON as its value. More information about environment variable groups can be found [here](https://docs.codemagic.io/variables/environment-variable-groups/). 

5. Go to your Google Cloud Platform console and open Cloud Storage. Make a note of the name of the bucket you want to upload to or create a new bucket.

> 6. Use the following script in your post-publishing script or create a new script step if you are using codemagic.yaml:

Reviewer: ```suggestion
6. Replace `gs://YOUR_BUCKET_NAME` with your own bucket name in the following script and add it as a post-publishing script or create a new script step if you are using codemagic.yaml:
```
